assets/app/features/upgrade/upgrade.controller.js

```diff
@@ -59,22 +37,16 @@
             if (vm.steps.isLastStep()) {
                 return;
             }
-            vm.steps.activeStep.active = false;
-            vm.steps.activeStep.enabled = true;
-
-            vm.steps.activeStep = vm.steps.list[vm.steps.activeStep.id + 1];
-            vm.steps.activeStep.active = true;
-            vm.steps.activeStep.enabled = true;
 
-            $state.go(vm.steps.activeStep.state);
+            $state.go(vm.steps.list[upgradeStepsFactory.activeStep.id + 1].state);
```

Comment:
The whole `nextStep()` and `isLastStep()` belong in `upgradeStepsFactory`. The factory already uses `$state` and owns the steps list. They can be linked here just like the rest of `vm.steps`.